src/test/integration/tokens.e2e-spec.ts

```diff
@@ -111,4 +123,73 @@ describe('Token Service', () => {
       expect(await tokenService.getToken(tokenIdentifier + 'a')).toBeUndefined();
     });
   });
-});
+
+  describe('Get Token Roles', () => {
+    it(`should return token roles`, async () => {
+      const tokenRoles = await tokenService.getToken(tokenIdentifier);
+
+      if (tokenRoles) {
+        const value = await tokenService.getTokenRoles(tokenRoles.identifier);
+        expect(value).toBeInstanceOf(Array);
+      }
+    });
+  });
+
+  describe('Get Token Accounts', () => {
+    it(`should return token with size of 10`, async () => {
+      const tokenRoles = await tokenService.getToken(tokenIdentifier);
+
+      if (tokenRoles) {
+        const tokensList = await tokenService.getTokenAccounts({from: 0, size: 10}, tokenRoles.name);
+        expect(tokensList).toBeInstanceOf(Array);
+      }
+    });
+  });
+
+  describe('Get Token For Address', () => {
+    it(`should return Token for address`, async () => {
+      const tokenFilter = new TokenFilter();
+      tokenFilter.name = 'MSFT-532e00';
+      const tokensList = await tokenService.getTokenForAddress(accountAddress, tokenFilter.name);
+
+      expect(tokensList).toBeInstanceOf(TokenWithBalance);
+    });
+    it(`should return undefined`, async () => {
+      const tokenFilter = new TokenFilter();
+      tokenFilter.name = 'invalidToken';
+      const tokensList = await tokenService.getTokenForAddress(accountAddress, tokenFilter.name);
+      expect(tokensList).toBeUndefined();
+    });
+  });
+
+  describe('Get All Tokens For Address', () => {
+    it(`should return all token for address`, async () => {
```

Comment:
This test wil allways return 2 tokens, due to filter.identifiers

Reply:
Change request resolved